java/fury-core/src/main/java/org/apache/fury/serializer/collection/AbstractMapSerializer.java

```diff
@@ -38,188 +39,296 @@
-        Object key = entry.getKey();
-        Object value = entry.getValue();
-        mapChunkWriter = mapChunkWriter.next(entry.getKey(), entry.getValue(), buffer);
-        mapChunkWriter.writeFinalKey(key, buffer, keySerializer);
-        mapChunkWriter.writeFinalValue(value, buffer, valueSerializer);
-        mapChunkWriter.increaseChunkSize();
-    }
-    mapChunkWriter.writeHeader(buffer);
-  }
-
-  private void genericJavaWrite(Fury fury, MemoryBuffer buffer, Map map) {
-    Generics generics = fury.getGenerics();
-    GenericType genericType = generics.nextGenericType();
-    if (genericType == null) {
-      generalJavaChunkWrite(fury, buffer, map);
-    } else {
-      GenericType keyGenericType = genericType.getTypeParameter0();
-      GenericType valueGenericType = genericType.getTypeParameter1();
-      // type parameters count for `Map field` will be 0;
-      // type parameters count for `SubMap<V> field` which SubMap is
-      // `SubMap<V> implements Map<String, V>` will be 1;
-      if (genericType.getTypeParametersCount() < 2) {
-        Tuple2<GenericType, GenericType> kvGenericType = getKVGenericType(genericType);
-        if (keyGenericType == objType && valueGenericType == objType) {
-          generalJavaChunkWrite(fury, buffer, map);
-          return;
-        }
-        keyGenericType = kvGenericType.f0;
-        valueGenericType = kvGenericType.f1;
-      }
-      // Can't avoid push generics repeatedly in loop by stack depth, because push two
-      // generic type changed generics stack top, which is depth index, update stack top
-      // and depth will have some cost too.
-      // Stack depth to avoid push generics repeatedly in loop.
-      // Note push two generic type changed generics stack top, which is depth index,
-      // stack top should be updated when using for serialization k/v.
-      // int depth = fury.getDepth();
-      // // depth + 1 to leave a slot for value generics, otherwise value generics will
-      // // be overwritten by nested key generics.
-      // fury.setDepth(depth + 1);
-      // generics.pushGenericType(keyGenericType);
-      // fury.setDepth(depth);
-      // generics.pushGenericType(valueGenericType);
-      boolean keyGenericTypeFinal = keyGenericType.isMonomorphic();
-      boolean valueGenericTypeFinal = valueGenericType.isMonomorphic();
-      if (keyGenericTypeFinal && valueGenericTypeFinal) {
-          javaKVTypesFinalChunkWrite(fury, buffer, map, keyGenericType, valueGenericType, generics);
-      } else if (keyGenericTypeFinal) {
-          javaKeyTypeFinalChunkWrite(fury, buffer, map, keyGenericType, valueGenericType, generics);
-      } else if (valueGenericTypeFinal) {
-          javaValueTypeFinalChunkWrite(fury, buffer, map, keyGenericType, valueGenericType, generics);
-      } else {
-          javaKVTypesNonFinalChunkWrite(fury, buffer, map, keyGenericType, valueGenericType, generics);
-      }
-    }
-  }
+    protected MethodHandle constructor;
```

Comment:
Could you format code by `mvn spotless:apply`? Those changes can then be removed

Reply:
done!
Spotless version 2.41.1 does not support JDK 8, at least JDK version 11 is required

Reply:
Yes, spotless needs jdk11 now